base/pkg/cache.jl

```diff
@@ -25,25 +25,41 @@ function mkcachedir()
     end
 end
 
-
 function prefetch(pkg::AbstractString, url::AbstractString, sha1s::Vector)
     isdir(".cache") || mkcachedir()
     cache = path(pkg)
-    if !isdir(cache)
+    repo = if isdir(cache)
+        try
+            LibGit2.GitRepo(cache) # open repo, free it at the end
+        catch err
+            rethrow(err)
+        end
+    else
         info("Cloning cache of $pkg from $url")
-        try Git.run(`clone -q --mirror $url $cache`)
-        catch
-            rm(cache, recursive=true)
-            rethrow()
+        try
+            # clone repo, free it at the end
+            LibGit2.clone(url, cache, bare = true, remote_cb = LibGit2.mirror_cb)
+        catch err
+            isdir(cache) && rm(cache, recursive=true)
+            error("Cannot clone $pkg from $url\n", err)
```

Comment:
+1 for PkgError, could help disambiguate exceptions if someone wants to script Pkg in clever ways